sdk/include/reactos/shlwapi_undoc.h

```diff
@@ -122,6 +124,56 @@ HRESULT WINAPI SHPropertyBag_WritePOINTL(IPropertyBag *ppb, LPCWSTR pszPropName,
 HRESULT WINAPI SHPropertyBag_WritePOINTS(IPropertyBag *ppb, LPCWSTR pszPropName, const POINTS *ppts);
 HRESULT WINAPI SHPropertyBag_WriteRECTL(IPropertyBag *ppb, LPCWSTR pszPropName, const RECTL *prcl);
 
+static inline
+VOID WINAPI
+SHPropertyBag_ReadIntDef(
+    IPropertyBag *ppb,
+    LPCWSTR pszPropName,
+    LPINT pnValue,
+    INT nDefaultValue)
+{
+    HRESULT hr = SHPropertyBag_ReadInt(ppb, pszPropName, pnValue);
+    if (FAILED(hr))
+        *pnValue = nDefaultValue;
+}
+
+static inline
+VOID WINAPI
+SHPropertyBag_ReadDWORDDef(
+    IPropertyBag *ppb,
+    LPCWSTR pszPropName,
+    LPDWORD pdwValue,
+    DWORD dwDefaultValue)
+{
+    HRESULT hr = SHPropertyBag_ReadDWORD(ppb, pszPropName, pdwValue);
+    if (FAILED(hr))
+        *pdwValue = dwDefaultValue;
+}
+
+static inline
+VOID WINAPI
+SHPropertyBag_ReadBOOLDef(
+    IPropertyBag *ppb,
+    LPCWSTR pszPropName,
+    LPBOOL pbValue,
+    BOOL bDefaultValue)
+{
+    HRESULT hr = SHPropertyBag_ReadBOOL(ppb, pszPropName, pbValue);
+    if (FAILED(hr))
+        *pbValue = bDefaultValue;
+}
+
+static inline
+BOOL WINAPI
+SHPropertyBag_ReadBOOLDefRet(
```

Comment:
Where have you found the existence of these values? They are definitively not in the PSDK, nor exported as such by shell32 (nor by ordinal only).

Reply:
From symbol files.